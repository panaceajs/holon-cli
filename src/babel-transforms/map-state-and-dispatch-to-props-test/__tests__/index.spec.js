const pluginTester = require('babel-plugin-tester');
const mapStateAndDispatchToPropsTest = require('../');
const pluginSyntaxObjectRestSpread = require('@babel/plugin-syntax-object-rest-spread');
const pluginJsx = require('@babel/plugin-syntax-jsx');
const dynamicImport = require('@babel/plugin-syntax-dynamic-import');

pluginTester({
  filename: __filename,
  plugin: mapStateAndDispatchToPropsTest,
  snapshot: true,
  babelOptions: {
    sourceType: 'module',
    plugins: [dynamicImport, pluginJsx, pluginSyntaxObjectRestSpread]
  },
  tests: [
    {
      title: 'it should change nothing when there is nothing to change.',
      fixture: '__fixtures__/with-test.js',
      snapshot: false
    },
    {
      title: 'it should add prop function tests.',
      fixture: '__fixtures__/with-test.js',
      pluginOptions: {
        actionTypes: {
          nonExistingAction: 'nonExistingActionValue'
        }
      }
    },
    {
      title: 'it should complete prop function tests.',
      fixture: '__fixtures__/with-test.js',
      pluginOptions: {
        actionTypes: {
          nonExistingAction: 'nonExistingActionValue',
          existingAction: 'existingActionValue'
        }
      }
    },
    {
      title: 'it should add prop tests.',
      fixture: '__fixtures__/with-test.js',
      pluginOptions: {
        mapProps: ['nonExistingProp']
      }
    },

    {
      title: 'it should complete prop tests.',
      fixture: '__fixtures__/with-test.js',
      pluginOptions: {
        mapProps: ['nonExistingProp', 'existingProp']
      }
    }
    // {
    //   title:
    //     'it should add action types to new state namespace in case of variable assignment',
    //   fixture: '__fixtures__/with-test.js',
    //   pluginOptions: {
    //     actionTypes: { foo: 'foo' },
    //     stateNamespace: 'nonExistingNamespace',
    //     variableName: 'initialState'
    //   }
    // }
  ]
});
