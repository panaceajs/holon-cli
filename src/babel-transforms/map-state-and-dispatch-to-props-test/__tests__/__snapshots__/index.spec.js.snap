// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`holon-map-state-and-dispatch-to-props-test it should add prop function tests.: it should add prop function tests. 1`] = `
"
describe('', () => {
  it('maps state and dispatch to props', () => {
    const wrapper = shallow(<Component store={store} />, {
      dive: true
    });

    // 
    expect(wrapper.props()).toEqual(expect.objectContaining({
      existingAction: 'existingActionValue',
      existingProp: 'existingPropShouldNotChange'
    }));
  });
});

      ↓ ↓ ↓ ↓ ↓ ↓

describe('', () => {
  it('maps state and dispatch to props', () => {
    const wrapper = shallow(<Component store={store} />, {
      dive: true
    });

    // 
    expect(wrapper.props()).toEqual(expect.objectContaining({
      existingAction: 'existingActionValue',
      existingProp: 'existingPropShouldNotChange',
      nonExistingAction: expect.any(Function),
      nonExistingActionValue: 'nonExistingActionValue-initial-value'
    }));
  });
});
"
`;

exports[`holon-map-state-and-dispatch-to-props-test it should add prop tests.: it should add prop tests. 1`] = `
"
describe('', () => {
  it('maps state and dispatch to props', () => {
    const wrapper = shallow(<Component store={store} />, {
      dive: true
    });

    // 
    expect(wrapper.props()).toEqual(expect.objectContaining({
      existingAction: 'existingActionValue',
      existingProp: 'existingPropShouldNotChange'
    }));
  });
});

      ↓ ↓ ↓ ↓ ↓ ↓

describe('', () => {
  it('maps state and dispatch to props', () => {
    const wrapper = shallow(<Component store={store} />, {
      dive: true
    });

    // 
    expect(wrapper.props()).toEqual(expect.objectContaining({
      existingAction: 'existingActionValue',
      existingProp: 'existingPropShouldNotChange',
      nonExistingProp: 'nonExistingProp-initial-value'
    }));
  });
});
"
`;

exports[`holon-map-state-and-dispatch-to-props-test it should complete prop function tests.: it should complete prop function tests. 1`] = `
"
describe('', () => {
  it('maps state and dispatch to props', () => {
    const wrapper = shallow(<Component store={store} />, {
      dive: true
    });

    // 
    expect(wrapper.props()).toEqual(expect.objectContaining({
      existingAction: 'existingActionValue',
      existingProp: 'existingPropShouldNotChange'
    }));
  });
});

      ↓ ↓ ↓ ↓ ↓ ↓

describe('', () => {
  it('maps state and dispatch to props', () => {
    const wrapper = shallow(<Component store={store} />, {
      dive: true
    });

    // 
    expect(wrapper.props()).toEqual(expect.objectContaining({
      existingAction: 'existingActionValue',
      existingProp: 'existingPropShouldNotChange',
      nonExistingAction: expect.any(Function),
      nonExistingActionValue: 'nonExistingActionValue-initial-value',
      existingActionValue: 'existingActionValue-initial-value'
    }));
  });
});
"
`;

exports[`holon-map-state-and-dispatch-to-props-test it should complete prop tests.: it should complete prop tests. 1`] = `
"
describe('', () => {
  it('maps state and dispatch to props', () => {
    const wrapper = shallow(<Component store={store} />, {
      dive: true
    });

    // 
    expect(wrapper.props()).toEqual(expect.objectContaining({
      existingAction: 'existingActionValue',
      existingProp: 'existingPropShouldNotChange'
    }));
  });
});

      ↓ ↓ ↓ ↓ ↓ ↓

describe('', () => {
  it('maps state and dispatch to props', () => {
    const wrapper = shallow(<Component store={store} />, {
      dive: true
    });

    // 
    expect(wrapper.props()).toEqual(expect.objectContaining({
      existingAction: 'existingActionValue',
      existingProp: 'existingPropShouldNotChange',
      nonExistingProp: 'nonExistingProp-initial-value'
    }));
  });
});
"
`;
